### DISCUSSION TITLE
How can i chnage hanging RKE2 Loglevel

### DISCUSSION THREAD

**Discussion by smartlocus** [rancher/rke2#4029] [Q&A]
**Environmental Info:**
RKE2 Version:  rke2 version v1.24.10+rke2r1 (1ccdce2571291649b9414af1f269f645c3fe4002)
go version go1.19.5 X:boringcrypto

Node(s) CPU architecture, OS, and Version: 

Cluster Configuration: currently using only the default master node

**Describe the bug:**  I am triyng to chnage the log level of my rke2 cluster to "Warning" since the default  info log level is consuming too much of my disk storage. How can i chnage the log level to "Warnings" for example in order to save disk usage ?

**Steps To Reproduce:**

- Installed RKE2: 

**Expected behavior:**

**Actual behavior:**

**Additional context / logs:**

**Comment by brandond** [comment]
Which component are you trying to change the log level of? The main rke2 service logs shouldn't consume much space at all, they're generally pretty low volume once the cluster is up and running.

**Reply by smartlocus** [reply]
![Screenshot from 2023-03-17 23-27-38](https://user-images.githubusercontent.com/114703928/226064141-a2326767-c375-45cb-a99e-b04d93f48ffb.png)


@brandond  i am having a high I/O. Can i reduce  that somehow? The rke2 server is somehow having a high writing to my disk continously

**Reply by brandond** [reply]
Etcd is very io intensive. If that is more than you can tolerate, you might try a single-server k3s cluster, which uses sqlite instead.

**Reply by smartlocus** [reply]
@brandond Could you explain me why the etcd of the rke2 server  is I/O intensive? I am trying to understand it 🙂

And also how can I change the log level of containers on my rke2 cluster. I searched for the rke2 -server.service file where I could change the log level to warning for example  but I could not find such a file and also did not find anything regarding loglevel in the documentation

**Reply by brandond** [reply]
> Could you explain me why the etcd of the rke2 server is I/O intensive?

Because it calls fsync to flush all pending IO to disk as part of every write to the datastore.

> And also how can I change the log level of containers on my rke2 cluster.

There is no way to globally change the log level of everything running in the cluster. Each pod is it own application, with its own command line flags and options. You'd have to pursue each of those individually.
